docs/ops/generation/RandomUniform_8.md

```diff
@@ -0,0 +1,246 @@
+```
+where x is uint32 generated random value.
+
+Double is formatted as follows: *sign*(1 bit) *exponent*(11 bits) *mantissa*(52 bits). The value is interpreted using following formula:
+\f[
+(-1)^{sign} * 1, mantissa * 2 ^{exponent - 1023}
+\f]
+
+so to obtain double values *sign*, *exponent* and *mantissa* are set as follows:
+``` 
+sign = 0
+exponent = 1023 - representation of a zero exponent.
+mantissa = 52 right bits from two concatinated uint32 values from random integer generator.
+``` 
+
+So the resulting double is obtained as follows:
+``` 
+mhi = x0 & 0xfffffu;  // upper 20 bits of mantissa
+mlo = x1;             // lower 32 bits of mantissa
+man = (mhi << 32) | mlo;  // mantissa
+val = ((exponent << 52) | man) - 1.0,
+```
+where x0, x1 are uint32 generated random values.
+
+To obtain a value in a specified range each value is processed with the following formulas:
+
+For float values:
+\f[
+result = x * (maxval - minval) + minval,
+\f]
+where *x* is random float or double value between [0..1).
+
+For integer values:
+\f[
+result = x \mod (maxval - minval) + minval,
+\f]
+where *x* is uint32 random value.
+
+
+Example 1. *RandomUniform* output with `seed` = 150, `seed2` = 10, `output_type` = f32:
+
+``` 
+input_shape    = [ 3, 3 ]
+output  = [[0.7011236  0.30539632 0.93931055]
+          [0.9456035   0.11694777 0.50770056]
+          [0.5197197   0.22727466 0.991374  ]]
+```
+
+Example 2. *RandomUniform* output with `seed` = 80, `seed2` = 100, `output_type` = double:
+
+``` 
+input_shape    = [ 2, 2 ]
+
+minval = 2
+
+maxval = 10
+
+output  = [[5.65927959 4.23122376]
+          [2.67008206 2.36423758]]
+```
+
+Example 3. *RandomUniform* output with `seed` = 80, `seed2` = 100, `output_type` = i32:
+
+``` 
+input_shape    = [ 2, 3 ]
+
+minval = 50
+
+maxval = 100
+
+output  = [[65 70 56]
+          [59 82 92]]
+```
+
+**Attributes**:
+
+* *output_type*
+
+    * **Description**: the type of the output.
+    * **Range of values**: "i32", "i64", "f16", "f32", "f64".
+    * **Type**: string
+    * **Default value**: "f32"
+    * **Required**: *No*
+
+* *seed*
+
+    * **Description**: global seed value.
+    * **Range of values**: positive integers
+    * **Type**: `int`
+    * **Default value**: None
+    * **Required**: *Yes*
+
+* *seed2*
+
+    * **Description**: operational seed value.
+    * **Range of values**: positive integers
+    * **Type**: `int`
+    * **Default value**: None
+    * **Required**: *Yes*
+
+**Inputs**:
+
+*   **1**: `shape` - 1D tensor of type *T_SHAPE* describing output shape. **Required.**
+
+*   **2**: `minval` - Scalar with type specified by the attribute *output_type*, defines the lower bound on the range of random values to generate (inclusive). Default value = 0.
+
+*   **3**: `maxval` - Scalar with type specified by the attribute *output_type*, defines the upper bound on the range of random values to generate (exclusive). Required for int64 and int32. Default value = 1.
```

Comment:
It is a bit painful when we have optional inputs. It is not easy to express in C++ code. So I would consider to make these inputs required. Let's discuss this during the online review.